server/athenian/api/internal/logical_repos.py

```diff
@@ -43,3 +47,52 @@ def extract_logical_repo(repo: str, offset: int = 2) -> str:
         return repo.split("/", offset)[offset]
     except IndexError:
         return ""
+
+
+@dataclasses.dataclass(frozen=True)
+class RepoName:
+    """The name of a repository.
+
+    Possible formats:
+
+    - repository name: "owner/reponame"
+    - logical repository name: "owner/reponame/logicalname".
+    - prefixed repository name: "provider.com/owner/reponame"
+    - prefixed logical repository name: "provider.com/owner/reponame/logicalname"
+
+    """
+
+    prefix: Optional[str]
+    organization: str
```

Comment:
```suggestion
    owner: str
```
(already used term in metadata DB / GraphQL)